xinference/model/llm/llm_family.py

```diff
@@ -251,7 +251,11 @@ def cache(
         llm_spec.model_size_in_billions,
         quantization,
     )
-    if os.path.exists(legacy_cache_path):
+    cache_dir = _get_cache_dir(llm_family, llm_spec, create_if_not_exist=False)
+    if os.path.exists(cache_dir):
+        logger.info("Cache path exists: %s", cache_dir)
+        return cache_dir
```

Comment:
Actually, In the following code （line 271 and line 274, function cache_from_huggingface and cache_from_modelscope）, we will also determine if the cache dir exists, which seems to duplicate your logic. See function `_skip_download`.